Create SeaIceSeasonality_DFA.ipynb

lidefi87 opened the pull request:
Submitting `SeaIceSeasonality` jupyter notebook to be included as cookbook example. This notebook calculates sea ice advance, retreat, and season duration, which are saved as netcdf files, based on daily sea ice concentration `ACCESS-OM2-01` outputs. The script also produces a composite figure with the three previously mentioned outputs. Calculations are based on Masson et al 2013 [doi:10.1371/journal.pone.0064756](url).

navidcy commented:
@aekiss any chance you can review this?

aekiss commented:
yes, I'll have a chance next week

navidcy commented:
@aekiss just a friendly reminder for this :)

aekiss commented:
Thanks @lidefi87, apologies for the long delay. 

I haven't looked at it thoroughly but I spotted a few things (only 2. looks serious):
1. if you omit the arguments to `Client` I think it will default to the number of cores available? If so, that will be more general.
1. the last line of `getACCESSdata` looks like it should be replaced by `vararray = vararray.sel(xt_ocean = slice(minlon, maxlon)).sel(yt_ocean = slice(minlat, maxlat))`, ie x and y are swapped
1. `str` not needed in 3rd code line of `SeaIceAdvArrays`?
1. can `getFileList` be replaced by calls to `glob` in package `glob`?
1. there are many places where you have hard-coded paths beginning `/g/data/v45/la6889` which won't work for anyone else. It would be easier to change for other users if `/g/data/v45/la6889` was defined as a variable, with the other paths formed by construction. Similarly, the repeated subdirectories could also be defined in one place as a variable. Also many of these paths are assumed to exist - it might be prudent to use `os.makedirs(path, exist_ok=True)` to guarantee it

navidcy commented:
@lidefi87, just checking. Would you be able to address @aekiss remarks? Let us know if you need any clarification or help with anything. Cheers!

lidefi87 commented:
Hi @navidcy! Thanks for the reminder. I got caught up with other tasks and this slipped my mind, but I have updated the script as per @aekiss comments above. 
1. The `n_workers` argument was removed from `Client`. The cores default to maximum available, but workers go down to 4. 
2. The lat and lon naming has been corrected.
3. Yes `str` is needed because the year information is used as part of a label later
4. I tried to do something similar with `glob`, but I cannot manage to get regex to work together with `glob`. But the `getFileList` also divides the file list into three sets of file paths depending on the file names.
5. The hard coded sections were only included as inputs for the functions defined in the script. These are meant to be updated by whoever is using the code without breaking the function itself. So that it is clear, I have updated all these sections with a label of `'PROVIDE_FILE_PATH'` so users are aware that they need to provide a file path of their own. I have now implemented the use of `os.makedirs(path, exist_ok=True)` in all functions that save outputs to disk. 

Let me know if there are any other things that need to be corrected.

Thanks,

Denisse

aekiss commented:
Great, thanks @lidefi87 I'll merge that.

lidefi87 commented:
Excellent! Hopefully I will have a new notebook to share in the next month or so. I will be back in touch.

Denisse

aekiss commented:
Great! looking forward to it. Thanks for your contribution.
